### PR TITLE
Add xchange v2 ethereum trades

### PR DESCRIPTION
**Pull request by mike-x7f**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

- Add xchange_v2_ethereum_base_trades to dex/models/trades/ethereum schema file
- Add xchange_v2_ethereum_base_trades to dex_ethereum_base_trades.sql
- Add xchange_v2 XchangePair_evt_Swap and XchangeFactory_evt_PairCreated to Ethereum sources yml file
- Add xchange_v2_ethereum_base_trades.sql leveraging xchange v2 factory pair created and swap data sources
- Add xchange v2 transactions to seed file

Thank you for your contribution!

### CODE REVIEW

**Comment by mike-x7f** [comment]
@jeff-dude is this conflict causing the test to not run? I tried adding in the latest valantis ref statement, but that didn't seem to work. Is there something you can do to resolve the conflict so I can see if the tests pass this time? Thanks, Mike

**Comment by jeff-dude** [comment]
> @jeff-dude is this conflict causing the test to not run? I tried adding in the latest valantis ref statement, but that didn't seem to work. Is there something you can do to resolve the conflict so I can see if the tests pass this time? Thanks, Mike

yes, CI is blocked if there are merge conflicts into main branch. i was able to resolve on the PR page itself. can you confirm the modified files still look as expected?

**Comment by jeff-dude** [comment]
are we blocked here? do we need @Hosuke to help take a look?

**Comment by mike-x7f** [comment]
Hi @jeff-dude @Hosuke sorry, we had a couple hurricanes come through where I live and it set me back a couple weeks. I will pick this up again soon. When I last worked on it, I was struggling to get the tests to run properly. If Hosuke wants to take a look at the previous runs and errors I was getting, that could be helpful. I tried running v1 and v2 together and separately and both were failing.

**Comment by jeff-dude** [comment]
> Hi @jeff-dude @Hosuke sorry, we had a couple hurricanes come through where I live and it set me back a couple weeks. I will pick this up again soon. When I last worked on it, I was struggling to get the tests to run properly. If Hosuke wants to take a look at the previous runs and errors I was getting, that could be helpful. I tried running v1 and v2 together and separately and both were failing.

glad you're safe!

@Hosuke if you don't mind, can you help here?

File: dbt_subprojects/dex/models/trades/ethereum/_schema.yml [hunk 1]
```diff
@@ -530,6 +530,23 @@ models:
...
+      tags: [ 'ethereum', 'dex', 'trades', 'xchange', 'uniswap', 'v2' ]
+    description: "xchange v2 ethereum base trades"
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - tx_hash
+            - evt_index
+      - check_dex_base_trades_seed:
+          seed_file: ref('xchange_ethereum_base_trades_seed')
+
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/ethereum/_schema.yml:549]:
```suggestion
      - check_dex_base_trades_seed:
          seed_file: ref('xchange_ethereum_base_trades_seed')
          filter:
            version: 2

```

File: sources/_sector/dex/trades/ethereum/_sources.yml [hunk 2]
```diff
@@ -113,6 +113,10 @@ sources:
     tables:
       - name: XchangePair_evt_Swap
       - name: XchangeFactory_evt_PairCreated
+  - name: xchange_v2_ethereum
+    tables:
+      - name: XchangePair_evt_Swap
+      - name: XchangeFactory_evt_PairCreated
```
Comment (Hosuke) [inline sources/_sector/dex/trades/ethereum/_sources.yml:119]: These sources look very similar...
We have two uniswap v2 forks xhange here.

**Comment by Hosuke** [comment]
I think all seed records might be covered in one base_trades spell.
I will do some experiments in this PR.

File: dbt_subprojects/dex/models/trades/ethereum/_schema.yml [hunk 3]
```diff
@@ -529,6 +529,27 @@ models:
...
+    description: "xchange v2 ethereum base trades"
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - tx_hash
+            - evt_index
+      - check_dex_base_trades_seed:
+          seed_file: ref('xchange_ethereum_base_trades_seed')
+          filter:
+            version: 2
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/ethereum/_schema.yml:552]:
```suggestion

  - name: xchange_v2_ethereum_base_trades
    meta:
      blockchain: ethereum
      sector: dex
      project: xchange
      contributors: mike-x7f, tomfutago
    config:
      tags: [ 'ethereum', 'dex', 'trades', 'xchange', 'uniswap', 'v2' ]
    description: "xchange v2 ethereum base trades"
    tests:
      - dbt_utils.unique_combination_of_columns:
          combination_of_columns:
            - tx_hash
            - evt_index
      - check_dex_base_trades_seed:
          seed_file: ref('xchange_ethereum_base_trades_seed')
```

File: sources/_sector/dex/trades/ethereum/_sources.yml (hunk 2 above)
Comment (Hosuke) [reply sources/_sector/dex/trades/ethereum/_sources.yml:119]: Seems they are different contracts:
[Xchange intersection test](https://dune.com/queries/4188446)

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @mike-x7f 

[Sample xchange_v2_ethereum_base_trades](https://dune.com/queries/4188573)

**Comment by mike-x7f** [comment]
> LGTM.✅
> 
> Thank you @mike-x7f 
> 
> 
> 
> [Sample xchange_v2_ethereum_base_trades](https://dune.com/queries/4188573)

Awesome! Thank you so much for the help!!

**Comment by 0xBoxer** [comment]
Due to a dbt version change, you need to merge main into your branch before the Github action runner runs successfully again.

**Review by jeff-dude — APPROVED** [review]
thanks both!

**Pull request merged**